packages/core/src/modules/credentials/services/CredentialService.ts

```diff
@@ -244,6 +244,24 @@ export class CredentialService {
       }),
     })
 
+    // Check if credential preview attributes match the schema attributes
+    const schema = await this.ledgerService.getSchema(credOffer.schema_id)
+    const credAttributes = credentialTemplate.preview.attributes.map((a) => a.name)
+    const schemaAttributes = schema.attrNames
+
+    const difference: string[] = []
+    credAttributes.forEach((credAtrr) => {
+      if (schemaAttributes.indexOf(credAtrr) === -1) {
+        difference.push(credAtrr)
+      }
+    })
+
+    if (difference.length > 0) {
+      throw new AriesFrameworkError(
+        `The credential preview attributes do not match the schema attributes (difference is: ${difference}, needs: ${schemaAttributes})`
+      )
+    }
```

Comment:
Might be good to extract this to a separate private method that is used in both createOffer and createOfferAsResponse to avoid duplication